src/plans/create/steps/general-information/PlanProjectField.tsx

```diff
@@ -0,0 +1,93 @@
+import { useFormContext, useWatch } from 'react-hook-form';
+
+import { useProjectNameSelectOptions } from '@components/common';
+import ControlledFormGroup from '@components/common/ControlledFormGroup';
+import { TypeaheadSelect } from '@components/common/TypeaheadSelect';
+import { V1beta1Provider } from '@kubev2v/types';
+import { useActiveNamespace } from '@openshift-console/dynamic-plugin-sdk';
+import { MenuToggleStatus } from '@patternfly/react-core';
+import { Namespace } from '@utils/constants';
+import { useForkliftTranslation } from '@utils/i18n';
+import { getDefaultNamespace } from '@utils/namespaces';
+
+import { GeneralFormFieldId, generalFormFieldLabels } from './constants';
+
+type PlanProjectFieldProps = {
+  onSelect: (value: string) => void;
+};
+
+export const PlanProjectField: FC<PlanProjectFieldProps> = ({ onSelect }) => {
+  const { t } = useForkliftTranslation();
+  const {
+    formState: { errors },
+    setValue,
+  } = useFormContext();
+  const [planProject, targetProject, targetProvider, sourceProvider]: [
+    string,
+    string,
+    V1beta1Provider,
+    V1beta1Provider,
+  ] = useWatch({
+    name: [
+      GeneralFormFieldId.PlanProject,
+      GeneralFormFieldId.TargetProject,
+      GeneralFormFieldId.TargetProvider,
+      GeneralFormFieldId.SourceProvider,
+    ],
+  });
+  const [activeNamespace] = useActiveNamespace();
+  const defaultNamespace = getDefaultNamespace();
+  const projectOptions = useProjectNameSelectOptions();
+
+  const initialProjectName = useMemo(
+    () =>
+      planProject ||
+      (activeNamespace === Namespace.AllProjects ? defaultNamespace : activeNamespace),
+    [defaultNamespace, activeNamespace, planProject],
+  );
+
+  return (
+    <ControlledFormGroup
+      isRequired
+      fieldId={GeneralFormFieldId.PlanProject}
+      label={generalFormFieldLabels[GeneralFormFieldId.PlanProject]}
+      controller={{
+        ...(projectOptions?.find((option) => option.value === initialProjectName) && {
+          defaultValue: initialProjectName,
+        }),
+        rules: { required: t('Plan project is required.') },
+        render: ({ field }) => (
+          <div ref={field.ref}>
+            <TypeaheadSelect
+              placeholder={t('Select plan project')}
+              id={GeneralFormFieldId.PlanProject}
+              selectOptions={projectOptions}
+              selected={field.value}
+              onSelect={(_, value) => {
+                field.onChange(value);
+                onSelect(value.toString());
+
+                if (sourceProvider) {
+                  setValue(GeneralFormFieldId.SourceProvider, '', { shouldValidate: true });
+                }
+                if (targetProvider) {
+                  setValue(GeneralFormFieldId.TargetProvider, '', { shouldValidate: true });
+                }
+                if (targetProject) {
+                  setValue(GeneralFormFieldId.TargetProject, '', { shouldValidate: true });
+                }
```

Comment:
im guessing you can omit the if checks and always clear the fields, they have a state of their own, so sending the same value wont make any difference.

Reply:
When running through the form the first time without values for these inputs already, the validation gets triggered, which I didn't think made sense since the user hasn't actually reached those fields yet. Does that make sense?

Here's a demo without these checks, which I don't think is ideal:

https://github.com/user-attachments/assets/6ce1b02c-1d03-479f-b819-01ca42b12236